libraries/chain/block_header_state.cpp

```diff
@@ -1,210 +1,95 @@
-         --i;
-         --blocks_to_confirm;
-      }
-
-      result.dpos_proposed_irreversible_blocknum = new_dpos_proposed_irreversible_blocknum;
-      result.dpos_irreversible_blocknum          = calc_dpos_last_irreversible(proauth.producer_name);
-
-      if (pending_schedule.schedule.producers.size() &&
-          result.dpos_irreversible_blocknum >= pending_schedule.schedule_lib_num) {
-         result.active_schedule = pending_schedule.schedule;
-
-         flat_map<account_name, uint32_t> new_producer_to_last_produced;
-
-         for (const auto& pro : result.active_schedule.producers) {
-            if (pro.producer_name == proauth.producer_name) {
-               new_producer_to_last_produced[pro.producer_name] = result.block_num;
-            } else {
-               auto existing = producer_to_last_produced.find(pro.producer_name);
-               if (existing != producer_to_last_produced.end()) {
-                  new_producer_to_last_produced[pro.producer_name] = existing->second;
-               } else {
-                  new_producer_to_last_produced[pro.producer_name] = result.dpos_irreversible_blocknum;
-               }
-            }
-         }
-         new_producer_to_last_produced[proauth.producer_name] = result.block_num;
-
-         result.producer_to_last_produced = std::move(new_producer_to_last_produced);
-
-         flat_map<account_name, uint32_t> new_producer_to_last_implied_irb;
-
-         for (const auto& pro : result.active_schedule.producers) {
-            if (pro.producer_name == proauth.producer_name) {
-               new_producer_to_last_implied_irb[pro.producer_name] = dpos_proposed_irreversible_blocknum;
-            } else {
-               auto existing = producer_to_last_implied_irb.find(pro.producer_name);
-               if (existing != producer_to_last_implied_irb.end()) {
-                  new_producer_to_last_implied_irb[pro.producer_name] = existing->second;
-               } else {
-                  new_producer_to_last_implied_irb[pro.producer_name] = result.dpos_irreversible_blocknum;
-               }
-            }
-         }
-
-         result.producer_to_last_implied_irb = std::move(new_producer_to_last_implied_irb);
-
-         result.was_pending_promoted = true;
-      } else {
-         result.active_schedule                                     = active_schedule;
-         result.producer_to_last_produced                           = producer_to_last_produced;
-         result.producer_to_last_produced[proauth.producer_name]    = result.block_num;
-         result.producer_to_last_implied_irb                        = producer_to_last_implied_irb;
-         result.producer_to_last_implied_irb[proauth.producer_name] = dpos_proposed_irreversible_blocknum;
-      }
-   } // !hotstuff_activated
-#endif
    
+   emplace_extension(result.header.header_extensions, instant_finality_extension::extension_id(),
+                     fc::raw::pack(instant_finality_extension{qc_info,
```

Comment:
std::move?

Reply:
The `qc_info` is just an `int` and a `bool`, so `std::move` doesn't make a difference, which is why I didn't add it.